lib/api/controllers/funnelController.js

```diff
@@ -127,7 +131,10 @@ class FunnelController {
 
     // resolves the callback immediately if a slot is available
     if (this.concurrentRequests < this.kuzzle.config.limits.concurrentRequests) {
-      return callback(null);
+      if (this.cachedIds[request.id]) {
+        delete this.cachedIds[request.id];
```

Comment:
as the following `return` statement is made out of the `if`, I cannot figure any situation where this would occur.
The `return true` will make the request being executed right away and it wont be added to the cache queue either (made line 159 below)